CommunicationLayer/CommDeviceControl/UdpMessageForwarder.cpp

```diff
@@ -26,27 +26,39 @@
 #include <QByteArray>
 #include <QDebug>
 #include <QHostAddress>
+#include <QTextCodec>
+
 
 #include "InfrastructureLayer/Settings/I_Settings.h"
 #include "UdpMessageForwarder.h"
 
+namespace
+{
+    const QString EXCHANGE_NAME = QString("amq.direct");
+}
+
 UdpMessageForwarder::UdpMessageForwarder(I_Settings& settings)
-: ipAddress_(settings.ipAddress())
-, port_(settings.udpPort())
 {
-    socket_.bind(ipAddress_, port_);
+  QString queueName = settings.queueName();
+
+  routingKey = settings.routingKey();
+
+  channel = Channel::Create(settings.ipAddress().toStdString(), (int)settings.udpPort());
+
+  channel->DeclareQueue(queueName.toStdString(), false, true, false, false);
+
+  channel->BindQueue(queueName.toStdString(), EXCHANGE_NAME.toStdString(), routingKey.toStdString());
+
 }
 
 UdpMessageForwarder::~UdpMessageForwarder()
 {
+  // TODO What to do for destructor? Nothing?
```

Comment:
That is my opinion too, it would delete the messages, but if we don't have a way to delete the queue built in here, and we don't have something built into the telemetry server, then the unconsumed messages will remain here until the queue is manually cleaned via the server commands. Otherwise the telemetry server will collect old data upon starting up